### PR TITLE
Add support for nv-ipam deployment

### PR DESCRIPTION
**Pull request by adrianchiris**
Cherry pick from master

### CODE REVIEW

File: deployment/network-operator/README.md [hunk 1]
```diff
@@ -540,13 +540,24 @@ optionally deployed components:
...
+| `ipamPlugin.imagePullSecrets` | list   | `[]`                           | An optional list of references to secrets to use for pulling any of the IPAM CNI Plugin image |
+
+#### NVIDIA IPAM Plugin
+
+| Name                      | Type   | Default            | description                                                                            |
+| ------------------------- | ------ | ------------------ | -------------------------------------------------------------------------------------- |
+| `nvIpam.deploy`           | bool   | `false`            | Deploy NVIDIA IPAM Plugin                                                              |
+| `nvIpam.image`            | string | `nvidia-k8s-ipam`  | NVIDIA IPAM Plugin image name                                                          |
+| `nvIpam.repository`       | string | `ghcr.io/mellanox` | NVIDIA IPAM Plugin image repository                                                    |
+| `nvIpam.version`          | string | `latest`           | NVIDIA IPAM Plugin image version                                                       |
```
Comment (moshe010) [inline deployment/network-operator/README.md:558]: we should update the version to the release one

File: pkg/state/state_nv_ipam_cni.go [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+//nolint:dupl
+func (s *stateNVIPAMCNI) Sync(customResource interface{}, _ InfoCatalog) (SyncState, error) {
+	cr := customResource.(*mellanoxv1alpha1.NicClusterPolicy)
+	log.V(consts.LogLevelInfo).Info(
+		"Sync Custom resource", "State:", s.name, "Name:", cr.Name, "Namespace:", cr.Namespace)
+
+	if cr.Spec.NvIpam == nil {
+		// Either this state was not required to run or an update occurred and we need to remove
+		// the resources that where created.
+		// TODO: Support the latter case
```
Comment (ykulazhenkov) [inline pkg/state/state_nv_ipam_cni.go:76]: Is this still the case?

Comment (adrianchiris) [reply pkg/state/state_nv_ipam_cni.go:76]: for 23.1 branch ? yes :)

**Comment by ykulazhenkov** [comment]
LGTM, just need to update IPAM version

**Comment by adrianchiris** [comment]
@moshe010 @ykulazhenkov  Done.

set nvidia ipam, version to v0.0.1.

if we end up updating it, a followup PR will be done.

**Review by ykulazhenkov — APPROVED** [review]

File: pkg/state/factory.go [hunk 3]
```diff
@@ -123,14 +123,19 @@ func newNicClusterPolicyStates(k8sAPIClient client.Client, scheme *runtime.Schem
...
+	}
 
 	return []Group{
 		NewStateGroup([]State{podSecurityPolicyState}),
 		NewStateGroup([]State{multusState, cniPluginsState, ipoibState, whereaboutState}),
 		NewStateGroup([]State{ofedState}),
 		NewStateGroup([]State{sriovDpState}),
 		NewStateGroup([]State{sharedDpState, nvPeerMemState}),
-		NewStateGroup([]State{ibKubernetesState}),
+		NewStateGroup([]State{ibKubernetesState, nvIpamCniState}),
```
Comment (tariq1890) [inline pkg/state/factory.go:138]: @adrianchiris Why add the nvIpamCNI to this stategroup? Shouldn't it belong to the multus & cniPlugins stategroup?

Comment (adrianchiris) [reply pkg/state/factory.go:138]: it doesnt really matter IMO, i have a refactor PR to remove the whole StateGroup stuff

**Pull request merged**